src/services/web3/web3.service.ts

```diff
@@ -1,20 +1,40 @@
-import { JsonRpcProvider } from '@balancer-labs/sor2/node_modules/@ethersproject/providers';
+import { ComputedRef, computed } from 'vue';
+import { Contract } from '@ethersproject/contracts';
+import { ErrorCode } from '@ethersproject/logger';
+import {
+  JsonRpcProvider,
+  TransactionResponse,
+  Web3Provider
+} from '@ethersproject/providers';
 import { rpcProviderService as _rpcProviderService } from '../rpc-provider/rpc-provider.service';
+import { logFailedTx } from '@/lib/utils/logging';
+import { gasPriceService } from '@/services/gas-price/gas-price.service';
+import ConfigService, { configService } from '@/services/config/config.service';
 
 interface Web3Profile {
   ens: string | null;
 }
 
+const RPC_INVALID_PARAMS_ERROR_CODE = -32602;
+const EIP1559_UNSUPPORTED_REGEX = /network does not support EIP-1559/i;
+
 export default class Web3Service {
-  provider: JsonRpcProvider;
+  provider: ComputedRef<Web3Provider | JsonRpcProvider>;
+
+  constructor(
+    private readonly rpcProviderService = _rpcProviderService,
+    private readonly config: ConfigService = configService
+  ) {
+    this.provider = computed(() => this.rpcProviderService.jsonProvider);
+  }
```

Comment:
We are mixing up providers here, lets be more explicit about them, there are two. The appProvider and the userProvider. The appProvider is for reading onchain data, the userProvider is for writing onchain data (signing transactions).

```suggestion
  appProvider: ComputedRef<JsonRpcProvider>;

  constructor(
    private readonly rpcProviderService = _rpcProviderService,
    private readonly config: ConfigService = configService
  ) {
    this.appProvider = computed(() => this.rpcProviderService.jsonProvider);
  }
```